book/src/generated/keymap.md

```diff
@@ -0,0 +1,429 @@
+| `k`, `up` | Extend up | `extend_line_up` |
+| `l`, `right` | Extend right | `extend_char_right` |
+| `w` | Extend to start of next word | `extend_next_word_start` |
+| `b` | Extend to start of previous word | `extend_prev_word_start` |
+| `e` | Extend to end of next word | `extend_next_word_end` |
+| `W` | Extend to start of next long word | `extend_next_long_word_start` |
+| `B` | Extend to start of previous long word | `extend_prev_long_word_start` |
+| `E` | Extend to end of next long word | `extend_next_long_word_end` |
+| `n` | Add next search match to selection | `extend_search_next` |
+| `N` | Add previous search match to selection | `extend_search_prev` |
+| `t` | Extend till next occurrence of char | `extend_till_char` |
+| `f` | Extend to next occurrence of char | `extend_next_char` |
+| `T` | Extend till previous occurrence of char | `extend_till_prev_char` |
+| `F` | Extend to previous occurrence of char | `extend_prev_char` |
+| `home` | Extend to line start | `extend_to_line_start` |
+| `end` | Extend to line end | `extend_to_line_end` |
+| `esc` | Exit selection mode | `exit_select_mode` |
+| `v` | Enter normal mode | `normal_mode` |
+## Unmapped Commands
+Some commands exist but do not have a default keybinding. These commands
+may be quite niche, or simply alternatives to the standard commands. If you
+want to use them, simply map them in your configuration (see [remapping]).
+
+[remapping]: ./remapping.md
+
+| Command | Description |
+| --- | --- |
+| `no_op` | Do nothing |
+| `move_prev_word_end` | Move to end of previous word |
+| `extend_line` | Select current line, if already selected, extend to another line based on the anchor |
+| `extend_line_above` | Select current line, if already selected, extend to previous line |
+| `goto_file_end` | Goto file end |
+| `goto_line_end_newline` | Goto line end |
+| `extend_to_line_end_newline` | Extend to line end |
+| `signature_help` | Show signature help |
+| `yank_joined_to_primary_clipboard` | Join and yank selections to primary clipboard |
+| `yank_main_selection_to_primary_clipboard` | Yank main selection to primary clipboard |
+| `replace_selections_with_primary_clipboard` | Replace selections by primary clipboard |
+| `paste_primary_clipboard_after` | Paste primary clipboard after selections |
+| `paste_primary_clipboard_before` | Paste primary clipboard before selections |
+
+ANCHOR_END: all
+ANCHOR: toc
+- [Normal](#normal)
+  - [Movement](#movement)
+  - [Changes](#changes)
+  - [Minor Modes](#minor-modes)
+    - [Goto](#goto)
+    - [Match](#match)
+    - [Left bracket](#left-bracket)
+    - [Right bracket](#right-bracket)
+    - [Window](#window)
+      - [New split scratch buffer](#new-split-scratch-buffer)
+    - [Space](#space)
+      - [Debug (experimental)](#debug-experimental)
+        - [Switch](#switch)
+    - [View](#view)
+  - [Selection Manipulation](#selection-manipulation)
+  - [Search](#search)
+  - [Shell](#shell)
```

Comment:
Used to be under Changes

Reply:
I feel like it's better as a separate section, as not all of the commands in shell are changes, for example `shell_keep_pipe` is about filtering selections, and not related to changes.